src/chainstate/stacks/boot/costs.clar

```diff
@@ -0,0 +1,495 @@
+;; the .costs contract
+
+;; Helper Functions
+
+;; Return a Cost Specification with just a runtime cost
+(define-private (runtime (r uint))
+    {
+        runtime: r,
+        write_length: u0,
+        write_count: u0,
+        read_count: u0,
+        read_length: u0,
+    })
+
+;; Linear cost-assessment function
+(define-private (linear (n uint) (a uint) (b uint))
+    (+ (* a n) b))
+
+;; TODO: fix this once log is available
```

Comment:
Is there an issue open somewhere for implementing integer log-2 as a Clarity function?

Actually, you could implement integer log-2 _in_ Clarity if you wanted -- all you'd need to do is find out the index of the highest bit set in a given uint (i.e. find the value for _i_ for a given _n_ where _(1 << (i + 1)) <= _n_ && _n_ <= (1 << i_)).

Reply:
Feel free to take inspiration from the `log2` Clarity implementation in ClarityTools by running the _Generate Clarity_ menu command on https://clarity.tools/code?KGxvZzIgOCk=

Reply:
https://github.com/blockstack/stacks-blockchain/issues/1862